File: Ej 18 (CarreraObstaculos)/Python/CarreraObstaculos.py
## DeathbatO

def paso_carrera(pista, atleta):
    if len(pista) == len(atleta):
        res = [""] * len(pista)
        for i in range(len(pista)):
            if (pista[i] == "_" and atleta[i] == "run") or (pista[i] == "|" and atleta[i] == "jump"):
                res[i] = pista[i]
            elif pista[i] == "_" and atleta[i] == "jump":
                res[i] = "x"
            elif pista[i] == "|" and atleta[i] == "run":
                res[i] = "/"
        
        resultado = res == pista
        print("Completo la carrera correctamente: " if resultado else "No completo la carrera correctamente: ")
        print(res)
        return resultado
    else:
        return False
    
p1 = ["_", "_", "|", "_", "_", "_", "|", "_"]
a1 = ["run", "run", "jump", "run", "run", "run", "jump", "run"]
p2 = ["_", "|", "_", "|", "_", "_", "|", "_"]
a2 = ["run", "jump", "run", "jump", "jump", "run", "run", "jump"]

print("Primera pista y recorrido: ")
print(p1)
print(a1, "\n")
paso_carrera(p1, a1)
print("\n\nSegunda pista y recorrido: ")
print(p2)
print(a2, "\n")
paso_carrera(p2, a2)

File: Ej 18 (CarreraObstaculos)/Kotlin/CarreraObstaculos.kt
// DeathbatO

import java.util.Arrays

fun pasoCarrera(pista: Array<String>, atleta: Array<String>): Boolean {
    if (pista.size == atleta.size) {
        val res = Array(pista.size) { "" }
        for (i in pista.indices) {
            when {
                (pista[i] == "_" && atleta[i] == "run") || (pista[i] == "|" && atleta[i] == "jump") ->
                    res[i] = pista[i]
                (pista[i] == "_" && atleta[i] == "jump") -> res[i] = "x"
                (pista[i] == "|" && atleta[i] == "run") -> res[i] = "/"
            }
        }
        val resultado = Arrays.equals(res, pista)
        println(if (resultado) "Completo la carrera correctamente: " else "No completo la carrera correctamente: ")
        println(Arrays.toString(res))
        return resultado
    } else {
        return false
    }
}

fun main() {
    val p1 = arrayOf("_", "_", "|", "_", "_", "_", "|", "_")
    val a1 = arrayOf("run", "run", "jump", "run", "run", "run", "jump", "run")
    val p2 = arrayOf("_", "|", "_", "|", "_", "_", "|", "_")
    val a2 = arrayOf("run", "jump", "run", "jump", "jump", "run", "run", "jump")

    println("Primera pista y recorrido: ")
    println(p1.contentToString())
    println(a1.contentToString() + "\n")
    pasoCarrera(p1, a1)
    println("\n\nSegunda pista y recorrido: ")
    println(p2.contentToString())
    println(a2.contentToString() + "\n")
    pasoCarrera(p2, a2)
}